src/Core/Geometry/MeshPrimitives.cpp

```diff
@@ -719,6 +726,7 @@ TriangleMesh makePlaneGrid( const uint rows,
     result.setVertices( std::move( vertices ) );
     result.setNormals( std::move( normals ) );
     result.setIndices( std::move( indices ) );
+    result.addAttrib( Ra::Engine::Data::Mesh::getAttribName( Ra::Engine::Data::Mesh::VERTEX_TEXCOORD ), std::move( texCoords ) );
```

Comment:
Instead of the dependence to Engine, use here "standard"  tex coord attrib name here : in_texcoord

Reply:
I can't do that, writing this seems to be a future delayed bug even if it is a standard name.
result.addAttrib( "in_texcoord", std::move( texCoords ) );
There are a duplication in the source code of the same information "in_texcoord". If one day a developer wants to change the names of shader attributes, it is highly probable that this one will be not aware to change this line too.
This is why I prefer to move the attribute name information to the Core. If you have any preferences for the location in the Radium hierarchy, I'm interested in it. For example in Core::Utils::Attribs (Attribs.hpp) or new Core::Mesh or new Core::Geometry::Mesh. For the moment I have positioned the update code in Core::Geometry (MeshPrimitives.hpp).
Thanks for your feedback.

Reply:
Even if it seems a good improvement, I think you can split the PR in two, since the proposal of using a const array in Core will break some external programs.
The second PR can fix generic attrib name management, and move it on the Core side. Then we can test it so one can update his own code before accepting the second PR.